### PR TITLE
[ADF-966] added the ability to reflect data change on datatable

### PR DESCRIPTION
**Pull request by VitoAlbano**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [X] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
The datatable won't refresh if the data passed when the table is not empty.
The datatable will be initialised only.

**What is the new behaviour?**
The datatable will refresh when a data object is passed

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: demo-shell-ng2/app/components/datatable/datatable-demo.component.html [hunk 1]
```diff
@@ -37,4 +37,5 @@
     <button md-raised-button (click)="addRow()">Add row</button>
     <button md-raised-button (click)="replaceRows()">Replace rows</button>
     <button md-raised-button (click)="replaceColumns()">Replace columns</button>
+    <button md-raised-button (click)="getRowForNode()">LOAD NODE</button>
```
Comment (eromano) [inline demo-shell-ng2/app/components/datatable/datatable-demo.component.html]: can you remove the uppercase?

File: ng2-components/ng2-alfresco-datatable/src/components/datatable/datatable.component.spec.ts [hunk 2]
```diff
@@ -66,6 +66,31 @@ describe('DataTable', () => {
         };
```
Comment (eromano) [inline ng2-components/ng2-alfresco-datatable/src/components/datatable/datatable.component.spec.ts]: can you remove the unnecessary indentation change?

Comment (VitoAlbano) [inline ng2-components/ng2-alfresco-datatable/src/components/datatable/datatable.component.spec.ts]: done

File: demo-shell-ng2/app/components/datatable/datatable-demo.component.ts [hunk 3]
```diff
@@ -172,4 +174,33 @@ export class DataTableDemoComponent {
...
+
+    getRowForNode() {
+        let opts: any = {
+            includeSource: true,
+            include: ['path', 'properties', 'allowableOperations']
+        };
+
+        Observable.fromPromise(this.apiService.getInstance().nodes
+            .getNodeInfo('-my-', opts)).subscribe((data) => {
+                console.log('FUnNy');
```
Comment (magemello) [inline demo-shell-ng2/app/components/datatable/datatable-demo.component.ts]: do we need this console log?

**Pull request merged**